src/core/__tests__/ReactErrorBoundaries-test.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+var React;
+var ReactDOM;
+
+describe('ReactErrorBoundaries', function() {
+
+  beforeEach(function() {
+    ReactDOM = require('ReactDOM');
+    React = require('React');
+  });
+
+  it('catches errors from children', function() {
+    var log = [];
+
+    class Box extends React.Component {
+      constructor(props) {
+        super(props);
+        this._errorMessage = undefined;
+      }
+      render() {
+        if (this._errorMessage) {
+          log.push('Box renderError');
+          return <div>Error: {this._errorMessage}</div>;
+        }
+        log.push('Box render');
+        var ref = function(x) {
+          log.push('Inquisitive ref ' + x);
+        };
+        return (
+          <div>
+            <Inquisitive ref={ref} />
+            <Angry />
+          </div>
+        );
+      }
+      handleError(e) {
+        this._errorMessage = e.message;
```

Comment:
`setState` does not work there.  Among other things, we rollback the transaction after calling handleError.  I'll try to figure out a way to make it work before merging.